src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskSerializer.scala

```diff
@@ -138,6 +149,24 @@ object TaskSerializer {
   }
 }
 
+object MarathonTaskStatusSerializer {
+  def fromProto(proto: Protos.MarathonTask.MarathonTaskStatus): MarathonTaskStatus = {
+    val result = MarathonTaskStatus.all.collectFirst {
+      case status: MarathonTaskStatus if status.toString == "" + proto => status
+    }
+    // TODO ju<>me discuss
+    result.orNull
```

Comment:
I'd make this more specific. The translation `MarathonTaskStatus -> Proto` can be compile safe via pattern matching; use that to construct a Mapping and the vice versa mapping. Fail if a proto value is not present in the mapping when doing `Proto -> MarathonTaskStatus`